src/wallet/wallet.cpp

```diff
@@ -2468,9 +2468,6 @@ CWallet::OutputAvailabilityResult CWallet::CheckOutputAvailability(
     // Check If not mine
     if (mine == ISMINE_NO) return res;
 
-    // Check if watch only utxo are allowed
-    if (mine == ISMINE_WATCH_ONLY && coinControl && !coinControl->fAllowWatchOnly) return res;
```

Comment:
Based on this change, we need to add a check in `StakeableCoins` after `CheckOutputAvailability`, to validate that the utxo is available, spendable and solvable before add it to the returned stakeable utxo vector.

Reply:
Inside `StakeableCoins` we call `CheckOutputAvailability` with `coinControl = nullptr`, so this line was already skipped before.
We should actually verify that the output is `spendable` (instead of just "available"), but that's regardless of this change.
I've added this in 9a8e3642089d29e3eb1ebeab3a7e6b7a2d2c1bcb.